crates/iceberg-catalog/src/api/management/v1/warehouse.rs

```diff
@@ -641,6 +656,108 @@ pub trait Service<C: Catalog, A: Authorizer, S: SecretStore> {
         Ok(())
     }
 
+    async fn undrop_tabulars(
+        request_metadata: RequestMetadata,
+        warehouse_ident: WarehouseIdent,
+        request: UndeleteTabularsRequest,
+        context: ApiContext<State<A, C, S>>,
+    ) -> Result<()> {
+        // ------------------- AuthZ -------------------
+        let catalog = context.v1_state.catalog;
+        let authorizer = context.v1_state.authz;
+        // TODO: authz
+        let mut futs = vec![];
+        match &request.undrop {
+            Undroppable::Tabulars { tabulars: tabs } => {
+                for t in tabs {
+                    match t {
+                        TabularIdentUuid::View(id) => {
+                            futs.push(authorizer.is_allowed_view_action(
+                                &request_metadata,
+                                warehouse_ident,
+                                (*id).into(),
+                                &crate::service::authz::CatalogViewAction::CanUndrop,
+                            ));
+                        }
+                        TabularIdentUuid::Table(id) => {
+                            futs.push(authorizer.is_allowed_table_action(
+                                &request_metadata,
+                                warehouse_ident,
+                                (*id).into(),
+                                &crate::service::authz::CatalogTableAction::CanUndrop,
+                            ));
+                        }
+                    }
+                }
+            }
+            Undroppable::Namespace { namespace: ns } => {
+                let false = authorizer
+                    .is_allowed_namespace_action(
+                        &request_metadata,
+                        warehouse_ident,
+                        (*ns).into(),
+                        &crate::service::authz::CatalogNamespaceAction::CanUndropAll,
+                    )
+                    .await?
+                else {
+                    return Err(ErrorModel::forbidden(
+                        "Not allowed to undrop all tabulars in the specified namespace.",
+                        "NotAuthorized",
+                        None,
+                    )
+                    .into());
+                };
+            }
+        }
+        if !futures::future::try_join_all(futs)
+            .await?
+            .into_iter()
+            .all(|t| t)
```

Comment:
in namespace case, we end up with empty futs, all on an empty iter evaluates to true